src/middleware/extract.company.number.middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { Request, Response, NextFunction } from "express";
+import { checkCompanyNumberFormatIsValidate } from "../utils/format/company.number.format";
+import { ContextKeys } from "../utils/constants/context.keys";
+import { getCompanyNumber } from "../utils/session";
+import { logger } from "../utils/logger";
+import { ValidateCompanyNumberFormat } from "../utils/validate/validate.company.number";
+
+declare module 'express-serve-static-core' {
+    interface Request {
+        companyNumber?: string;
+    }
+}
+
+export const extractCompanyNumberMiddleware = (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const isSessionSet = typeof req.session !== "undefined";
+        const companyNumberParam = req.params.companyNumber;
+        let companyNumber: string = "";
+
+        if (companyNumberParam) {
+            checkCompanyNumberFormatIsValidate(companyNumberParam);
+            companyNumber = companyNumberParam;
+
+            if (isSessionSet && ValidateCompanyNumberFormat.isValid(companyNumber)) {
```

Comment:
You're checking the company number is valid twice here, also why do we have two independent functions that do the same thing?